sui_core/src/authority/authority_store.rs

```diff
@@ -142,9 +149,12 @@ pub fn open_cf_opts<P: AsRef<Path>>(
     Ok(rocksdb)
 }
 
-impl<const ALL_OBJ_VER: bool> SuiDataStore<ALL_OBJ_VER> {
+impl<const ALL_OBJ_VER: bool, S: AuthoritySignInfoTrait> SuiDataStore<ALL_OBJ_VER, S>
+where
+    TransactionEnvelope<S>: PartialEq,
```

Comment:
If the bound on `S` is `: Eq + ...` (which implies a `PartialEq` bound), and `TransactionEnvelope` derives `PartialEq` and `Eq`, then you don't need a `PartialEq` bound on `TransactionEnvelope<S>`, it's satisfied by derivation.